src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs

```diff
@@ -74,41 +76,44 @@ public class VsCredentialProviderImporter
             _initializer = initializer ?? Initialize;
         }
 
-        [Import("VisualStudioAccountProvider", typeof(IVsCredentialProvider), AllowDefault = true)]
-        public IVsCredentialProvider ImportedProvider { get; set; }
+        [ImportMany(typeof(IVsCredentialProvider))]
+        public IEnumerable<IVsCredentialProvider> ImportedProviders { get; set; }
 
         /// <summary>
         /// Plugin providers are entered loaded the same way as other nuget extensions,
         /// matching any extension named CredentialProvider.*.exe.
         /// </summary>
         /// <returns>An enumeration of plugin providers</returns>
-        public ICredentialProvider GetProvider()
+        public IReadOnlyCollection<ICredentialProvider> GetProviders()
         {
             this._initializer();
-            ICredentialProvider result = null;
+            var results = new List<ICredentialProvider>();
 
-            if (ImportedProvider != null)
+            if (ImportedProviders != null)
             {
-                result = new VsCredentialProviderAdapter(ImportedProvider);
+                foreach (var importedProvider in ImportedProviders)
+                {
+                    results.Add(new VsCredentialProviderAdapter(importedProvider));
+                }
             }
 
             // Dev15+ will provide a credential provider for VSTS.
             // If we are in Dev14, and no imported VSTS provider is found,
             // then fallback on the built-in VisualStudioAccountProvider
-            if (result == null && IsDev14)
+            if (!results.Any() && IsDev14)
```

Comment:
nit: `results` is a `List` so you can use `Count` here rather then expensive Linq query